### PR TITLE
FIx publish in official builds

### PR DESCRIPTION
**Pull request by jkotas**
Update the platform matrix and build command line for official builds to match what it was before the template change

### CODE REVIEW

File: eng/pipelines/runtimelab.yml
Comment (jkotas) [inline eng/pipelines/runtimelab.yml:129]: I am updating the official builds to this

File: eng/pipelines/runtimelab.yml
Comment (jkotas) [inline eng/pipelines/runtimelab.yml:133]: And this

**Comment by jkotas** [comment]
@dotnet/nativeaot-llvm

**Review by akoeplinger — APPROVED** [review]

**Comment by yowl** [comment]
Thanks, I defer to others with better knowledge!

File: eng/pipelines/runtimelab-official.yml
Comment (SingleAccretion) [inline eng/pipelines/runtimelab-official.yml]: I don't think this will work - and I don't think the merged version of `runtimelab.yml` `isOfficialBuild` branch was quite correct either. The `host` subset is gated by `NativeAotSupported` (`false` for WASM).

Here's how this looked before the merge:

https://github.com/dotnet/runtimelab/blob/143b29c91ff4b62a0a93af6af4457f6127c7f8a1/eng/pipelines/runtimelab.yml#L110-L128

It is meant to work as follows:
1) We produce the target (WASM) packages in the first part of the build. Note: arch-independent packages are also produced here, but there is a later condition somewhere that [picks](https://github.com/dotnet/runtimelab/blob/143b29c91ff4b62a0a93af6af4457f6127c7f8a1/eng/pipelines/official/stages/publish.yml#L2) one platform out of which it is published.
2) We produce the host (`runtime.win-x64`) packages in the post-build steps. We only do this for `browser_wasm_win` to avoid the conflict.

The problem in https://github.com/dotnet/runtimelab/issues/2539#issuecomment-2080367183 is that we switched `windows_arm64` to `windows_x64` and so one of the target packages collided with one of the host packages. That is something I failed to consider with https://github.com/dotnet/runtimelab/pull/2563#discussion_r1580151005.

The correct fix is to probably just remove all but `Browser_wasm_win` and `wasi_wasm_win` from the platform list here (and assume that whatever breakage of upstream we cause will be caught by PR validation).

Comment (jkotas) [reply eng/pipelines/runtimelab-official.yml]: Thank you - this makes sense. I was puzzled by how this can ever work, but I did not consider that runtimelab.yml can have incorrectly resolved conflicts.

**Review by agocke — COMMENTED** [review]
Retroactive LGTM

**Pull request merged**
